tokio/src/runtime/thread_pool/owned.rs

```diff
@@ -1,12 +1,20 @@
-use crate::runtime::task::{self, Task};
+use crate::loom::sync::atomic::AtomicUsize;
 use crate::runtime::thread_pool::{queue, Shared};
+use crate::task::{self, Task};
 use crate::util::FastRand;
 
 use std::cell::Cell;
 
 /// Per-worker data accessible only by the thread driving the worker.
 #[derive(Debug)]
 pub(super) struct Owned<P: 'static> {
+    /// Worker generation. This guards concurrent access to the `Owned` struct.
+    /// When a worker starts running, it checks that the generation it has
+    /// assigned matches the current generation. When it does, the worker has
+    /// obtained unique access to the struct. When it fails, another thread has
+    /// gained unique access.
+    pub(super) generation: AtomicUsize,
```

Comment:
Used by `worker.rs` to obtain a lock on the owned state. When a worker is handed off to a new thread, the recipient must successfully CAS this value to +1. If it fails, the original worker reclaimed the state.